Expensify/App
# Display Full Name in LHN and Header if available

@marcaaron will you please review this?

Fixes https://github.com/Expensify/ReactNativeChat/issues/337

### Details
This PR displays each user's Full Name in the LHN/Header if available, and falls back to the email (current value).

### Tests
1. Log into an account and message a user that has their First/Last name set in personal details, confirm that the chat shows their full name in the LHN and in the report header
2. Message a user that doesn't have their First/Last name set in personal details, confirm that the chat shows the email in the LHN and report header
3. Open expensify.com.dev and create a 3 way chat by calling the following in the JS console. **Make sure you replace the emailList value with emails on your dev env**. The `console.log` just spits out the reportID.
```
API.createChatReport({emailList: 'test@test.com,test2@test.com,joe@expensify.com'}).done(function (response) { console.log(response); });
```
4. Login as one of the users on mobile, and another on web. Leave comments on both platforms and confirm that the Full Name appears if available, and if not the email fallback shows.
5. In the LHN search box, start typing. Confirm that the results are accurate, and that if the user has a full name it is displayed above the email, otherwise just the email is shown

### Screenshots
| Web | Mobile LHN | Mobile Header |
|---|---|---|
| ![image](https://user-images.githubusercontent.com/3981102/93839919-5fe3e280-fc43-11ea-8d0e-45313979f6b4.png) | ![image](https://user-images.githubusercontent.com/3981102/93839922-64a89680-fc43-11ea-969c-67ac944d35c0.png) | ![image](https://user-images.githubusercontent.com/3981102/93839934-6f632b80-fc43-11ea-9705-73c240157819.png) |
